plugins/scaffolder-react/src/next/components/Workflow/Workflow.tsx

```diff
@@ -74,9 +80,71 @@ export const Workflow = (workflowProps: WorkflowProps): JSX.Element | null => {
   });
 
   const errorApi = useApi(errorApiRef);
-
   const { loading, manifest, error } = useTemplateParameterSchema(templateRef);
 
+  const featureFlagKey = 'backstage:featureFlag';
+  const featureFlagApi = useApi(featureFlagsApiRef);
+
+  const filterOutProperties = useCallback(
+    (step: Step): Step => {
+      const filteredStep = cloneDeep(step);
+      const removedPropertyKeys: Array<string> = [];
+      if (filteredStep.schema.properties) {
+        filteredStep.schema.properties = Object.fromEntries(
+          Object.entries(filteredStep.schema.properties).filter(
+            ([key, value]) => {
+              if (value[featureFlagKey]) {
+                if (featureFlagApi.isActive(value[featureFlagKey])) {
+                  return true;
+                }
+                removedPropertyKeys.push(key);
+                return false;
+              }
+              return true;
+            },
+          ),
+        );
+
+        // remove the feature flag property key from required if they are not active
+        filteredStep.schema.required = Array.isArray(
+          filteredStep.schema.required,
+        )
+          ? filteredStep.schema.required?.filter(
+              r => !removedPropertyKeys.includes(r as string),
+            )
+          : filteredStep.schema.required;
+      }
+      return filteredStep;
+    },
+    [featureFlagKey, featureFlagApi],
+  );
+
+  const filterOutSteps = useCallback(
+    m => {
+      if (m) {
+        const filteredSteps = m.steps
+          .filter(step => {
+            const featureFlag = step.schema[featureFlagKey];
+            return (
+              typeof featureFlag !== 'string' ||
+              featureFlagApi.isActive(featureFlag)
+            );
+          })
+          .map(filterOutProperties);
+
+        m.steps = filteredSteps;
+      }
+
+      return m;
+    },
+    [featureFlagApi, filterOutProperties, featureFlagKey],
+  );
+
+  const sortedManifest = useMemo(
```

Comment:
I'd like to move all of this logic into a hook rather than polluting the workflow component here.

Something like `useStepProperties()` or something like that?

Maybe can workshop the name a little bit :joy:

Reply:
Thank you for the feedback! I will fix it soon.